src/std/ide/swank/autodoc.ss

```diff
@@ -0,0 +1,93 @@
+(import :gerbil/expander ./message :std/sugar
+	./context ./api :std/format
+	:std/srfi/13)
+(export #t)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Completions, Autodoc and Signature
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; search for a procedure in 0gambit-procedures
+;; returns the procedure symbol if it finds it
+
+(gxc#optimizer-info-init!)
+(def src (path-expand "src/" (gerbil-home)))
+
+(try
+ (gx#import-module
+  (path-expand "gerbil/builtin.ssxi.ss" src) #t #t)
+ (gx#import-module
+  (path-expand "gerbil/builtin-inline-rules.ssxi.ss" src) #t #t)
+ (let* ((lpath  (path-expand "lib/gerbil/runtime/" (gerbil-home)))
+	(fs (filter (cut string-suffix? ".ssxi.ss" <>)
+		    (directory-files lpath)))
+	(mods (map (cut path-expand <> lpath) fs)))
+   (for-each (cut gx#import-module <> #t #t) mods))
+ (catch (_) #f))
+
+(def (swank-pp-proc-args args)
+  (def (strim sym)
+    (string->symbol
+     (string-trim-both
+      (symbol->string sym)  
+      (? (or (cut char=? <> #\_) char-numeric?)))))
+  (let lp ((lst args))
+    (if (null? lst) lst
+	(with ([kar . kdr] lst)
+	  (cons (strim kar)
+		(if (pair? kdr) (lp kdr) (strim kdr)))))))
+
+(def (swank-autodoc-procedure sym proc)
+  (def name (##procedure-name proc))
+  (def opti (and name (gxc#optimizer-lookup-type name)))
+  (def sig (and opti (gxc#!procedure? opti) (gxc#!procedure-signature opti)))
+  (def ret (and sig (gxc#!signature? sig) (gxc#!signature-return sig)))
```

Comment:
here too and more below, i am going to stop.

Reply:
Yup, done?